### PR TITLE
fix(llm): stop sending default temperature/top_p to OpenAI reasoning models (task-404)

### PR DESCRIPTION
**Pull request by rmusser01**
## What

`chat_with_openai` resolved `temperature`/`top_p` through config defaults (0.7/0.95) and sent them on **every** request — OpenAI reasoning models (o-series, gpt-5 family) reject both with HTTP 400 ("Unsupported parameter"), so any call routed to them through this handler failed, on both the Chat Completions and Responses-API branches. Found during the prefill provider verification (task-403).

## Fix

- New `_is_openai_reasoning_model` predicate: o1/o3/o4/gpt-5 families, matched exactly or at a `-`/`.` boundary (so `o365-copilot` and `olmo-7b` never match).
- The sampling block is skipped when the model is reasoning-family **or** the request takes the Responses-API branch; explicit caller-passed values on that path are dropped with a warning (the API rejects any value, so sending them would just 400).
- Non-reasoning models keep byte-identical behavior (config defaults still included).

## Tests

17 new tests at the HTTP seam (fake `requests.Session` capturing the JSON payload): both branches omit sampling, explicit-value drop, preserved non-reasoning behavior, and a 13-case predicate boundary matrix. Handler-adjacent suites at their documented pre-existing baseline (6 pytest-mock errors in untouched files).

**Live proof:** the exact previously-400ing repro (`gpt-5-mini`, `reasoning_effort='low'` through `chat_api_call`) now succeeds against the real API.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---
## Summary by cubic
Stop sending default `temperature`/`top_p` to OpenAI reasoning models to prevent 400s, keeping sampling for other models. Addresses task-404; `gpt-5-*` and `o*-*` requests (including via the Responses API) now succeed.

- **Bug Fixes**
  - Added `_is_openai_reasoning_model` (o1/o3/o4, gpt-5) with case-insensitive hyphen/dot boundary matching.
  - Omit `temperature`/`top_p` for reasoning models and all Responses API calls; drop caller-provided values with a warning.
  - Preserved defaults for non-reasoning models; added 17 HTTP-seam tests across both paths and boundary cases.

- **Refactors**
  - Tests now capture posted payloads via a fixture closure, removing shared class state.

<sup>Written for commit 932c22bccc76d4adf881b056a5774063d75150aa. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/rmusser01/tldw_chatbook/pull/736?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: Tests/Chat/test_openai_reasoning_sampling_params.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+                "usage": {},
+            }
+        )
+
+
+@pytest.fixture
+def captured_payloads(monkeypatch):
+    _FakeSession.captured = []
+    monkeypatch.setattr(llm_calls.requests, "Session", _FakeSession)
+    return _FakeSession.captured
```
[the remark this answers is not in the document]
Comment (rmusser01) [reply Tests/Chat/test_openai_reasoning_sampling_params.py]: Accepted: fixture now defines the fake session inline and captures payloads via closure — no mutable class-level state (commit pushed).

File: tldw_chatbook/LLM_Calls/LLM_API_Calls.py [hunk 2]
```diff
@@ -198,6 +198,33 @@ def _openai_use_responses_api(
     )
 
 
+_OPENAI_REASONING_MODEL_FAMILIES = ("o1", "o3", "o4", "gpt-5")
+
+
+def _is_openai_reasoning_model(model: object) -> bool:
```
[the remark this answers is not in the document]
Comment (rmusser01) [reply tldw_chatbook/LLM_Calls/LLM_API_Calls.py:204]: Declined with evidence: `object` matches this file's local convention for defensive-input predicates — the sibling `_openai_use_responses_api(reasoning_effort: object, ...)` directly above uses the same hint. `object` is also the stricter choice here: the function's first act is normalization via `str()`, and `object` (unlike `Any`) prevents accidental attribute use without conversion.

File: tldw_chatbook/LLM_Calls/LLM_API_Calls.py [hunk 3]
```diff
@@ -533,11 +560,22 @@ def chat_with_openai(
         payload["input"] = api_messages
     else:
         payload["messages"] = api_messages
-    # Add optional parameters if they have a value
-    if final_temp is not None:
-        payload["temperature"] = final_temp
-    if final_top_p is not None:
-        payload["top_p"] = final_top_p  # OpenAI uses top_p
+    # Add optional parameters if they have a value. Reasoning-family models
```
[the remark this answers is not in the document]
Comment (rmusser01) [reply tldw_chatbook/LLM_Calls/LLM_API_Calls.py:563]: Declined with evidence, three parts:
1. The premise ('none' = off/disable in the UI) doesn't hold: `_REASONING_EFFORT_VALUES` in console_session_settings.py models the API's real effort tiers ({none, minimal, low, medium, high, xhigh}) — 'none' is a genuine tier on current gpt-5.x models, not a sentinel. The handler deliberately distinguishes the two cases already: reasoning_summary='none' IS special-cased as off (skipped when building reasoning_options), while effort='none' is forwarded as a real value. That asymmetry is intentional, pre-existing behavior.
2. The 'none'→Responses-API routing is untouched by this PR (`_openai_use_responses_api`/`_is_present_setting` are unchanged).
3. The new sampling omission is correct at every effort tier: OpenAI rejects temperature/top_p at the MODEL level for the reasoning family (the live 400 reads 'not supported with this model'), so a gpt-5.x call with effort='none' must still omit them. The residual scenario (a non-reasoning model with effort set, expecting 'off') errors at the provider identically pre- and post-PR — if the Console ever wants effort='none' to mean 'route as plain chat completions', that's a settings-semantics change to file separately, not a regression of this fix.
